fetcher.go

```diff
@@ -0,0 +1,148 @@
+package tokendirectory
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/0xsequence/go-sequence/lib/prototyp"
+	"github.com/rs/zerolog/log"
+)
+
+type Fetcher struct {
```

Comment:
rename this to just TokenDirectory .. even though it does fetching, I think model it against the "domain", which is that this object gives you a TokenDirectory interface